spring-cloud-sleuth-instrumentation/src/main/java/org/springframework/cloud/sleuth/instrument/batch/TraceJobExecutionListener.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2018-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.batch;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.JobExecutionListener;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.SpanAndScope;
+import org.springframework.cloud.sleuth.Tracer;
+
+class TraceJobExecutionListener implements JobExecutionListener {
+
+	private final Tracer tracer;
+
+	private static Map<JobExecution, SpanAndScope> SPANS = new ConcurrentHashMap<>();
+
+	TraceJobExecutionListener(Tracer tracer) {
+		this.tracer = tracer;
+	}
+
+	@Override
+	public void beforeJob(JobExecution jobExecution) {
+		Span span = this.tracer.nextSpan().name(jobExecution.getJobInstance().getJobName());
+		span.tag("job.name", jobExecution.getJobInstance().getJobName());
+		// TODO: How to add step type?
```

Comment:
@mminella mentioned some things about types of steps - I guess he meant the implementation type?

Reply:
We are in a `JobExecutionListener#beforeJob` here, so steps are not started yet at this point. So I'm not sure if it makes sense to look for a step (type) here.

Reply:
@jonatan-ivanov @shakuzen do you think the tag should be `batch.job.name` or `job.name` ?

Reply:
This is the hard part. :D
Things to consider:
- `job.name` is shorter and descriptive enough so that people will know what is it
- `batch.job.name` fits better to other things like `mvc.controller.class` and `mvc.controller.method`
- there might be tags that are outside of the job (`batch.xyz`)
- Micrometer [calls it](https://docs.spring.io/spring-batch/docs/4.3.x/reference/html/monitoring-and-metrics.html#built-in-metrics) `job.name`
- 🤯 

With this, I would vote for `job.name` but if you ask me why, I might change my mind. :D